Inc/HALALMock/Services/ADC/ADC.hpp

```diff
@@ -22,24 +23,31 @@ using std::string;
 #define LPTIM3_PERIOD 6875
 
 #define ADC_MAX_VOLTAGE 3.3
-#define MAX_12BIT 4095.0
-#define MAX_16BIT 65535.0
+#define MAX_16BIT 65535.0	// 2^16 - 1
+#define MAX_14BIT 16383.0	// 2^14 - 1
+#define MAX_12BIT 4095.0	// 2^12 - 1
+#define MAX_10BIT 1023.0	// 2^10 - 1
+
 
 
-/**
- * @brief A utility class that controls ADC inputs.
- *
- * This class handles the configuration, read, and correct casting of the ADC values in the ADC inputs using DMA to make cyclic reads, and the code pulls from the space given to the DMA memory.
- * For the class to work, the pins used as parameters in each inscribe has to be declared on Runes and on Pins as ADCs (change adc_instances and AF of pins, respectively).
- * Moreover, if the ADC used is not on the code generated by the .ioc you may need to change HAL_ADC_MspInit (on stm32h7xx_hal_msp.c) to configure the ADC pins and channels and their DMAs.
- * Template-project has all the adcs declared on runes and the .ioc correctly configured, so unless more are needed these configurations can be assumed.
- */
 class ADC {
 public:
+
+	/// In STM32H723ZG, the ADC1 and ADC2 has 16 bits as their maximum resolution,
+	/// while the ADC3 has 12 bits. Both of them can be configured to has less resolution than its maximum
+	enum ADCResolution {
+		ADC_RES_16BITS,
+		ADC_RES_14BITS,
+		ADC_RES_12BITS,
+		ADC_RES_10BITS
+	};
+
 	struct InitData {
 	public:
 		ADC_TypeDef* adc;
 		uint32_t resolution;
+		/// @brief Resolution of the emulated ADC translated to an enum
+		ADCResolution emulated_resolution;
```

Comment:
don't modify public interfaces, now anything that creates this is invalid because emulated_resolution has not been initialized.

Reply:
So how can I implement it? Typing it in private section would work?